src/errors/InvalidSearchParameterError.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+// eslint-disable-next-line import/prefer-default-export
+export class InvalidSearchParameterError extends Error {
```

Comment:
Worth-while to make this a generic `InvalidParameterError`?

Reply:
I like it better for it to be specific to search. It's ok to have several errors that map to 400, like `InvalidResourceError`